builder/common/step_security_group.go

```diff
@@ -194,16 +200,31 @@ func (s *StepSecurityGroup) Run(ctx context.Context, state multistep.StateBag) m
 
 	port := s.CommConfig.Port()
 	// Authorize access for the provided port within the security group
-	groupRules := &ec2.AuthorizeSecurityGroupIngressInput{
-		GroupId: groupResp.GroupId,
-		IpPermissions: []*ec2.IpPermission{
-			{
-				FromPort:   aws.Int64(int64(port)),
-				ToPort:     aws.Int64(int64(port)),
-				IpRanges:   groupIpRanges,
-				IpProtocol: aws.String("tcp"),
+	var groupRules *ec2.AuthorizeSecurityGroupIngressInput
+	if len(groupIpv6Ranges) != 0 {
```

Comment:
i am trying to understand the need for the `if` check here, why do we need to assign a security group rule that is either `ipv6` or `ipv4`?
For example if a user provides a `temporary_security_group_cidrs` as 
```
temporary_security_group_source_cidrs = ["0.0.0.0/16", "51.207.193.182/32", "::/0"]
```
Wouldnt we want to apply both the ipv4 and ipv6 CIDRs to the security group.

i.e. 
```suggestion
groupRules = &ec2.AuthorizeSecurityGroupIngressInput{
	GroupId: groupResp.GroupId,
	IpPermissions: []*ec2.IpPermission{
		{
			FromPort:   aws.Int64(int64(port)),
			ToPort:     aws.Int64(int64(port)),
			Ipv6Ranges: groupIpv6Ranges,
			IpRanges:   groupIpRanges,
			IpProtocol: aws.String("tcp"),
		},
	},
}
```